src/quilla/steps/steps_aggregator.py

```diff
@@ -84,11 +87,19 @@ def run_steps(self) -> List[BaseReport]:
         reports: List[BaseReport] = []
         for i, step in enumerate(self._steps):
             try:
+                self.ctx.logger.debug('Running step %s', step.action.value)
                 report = step.perform()
             except Exception as e:
                 if not self.ctx.suppress_exceptions:
```

Comment:
Two questions: 1. under which condition would we want to suppress exception? 2. if exception is raised on line 95, I don't think line 97-104 will be executed.

Reply:
In most conditions that Quilla runs in, the exception should be suppressed because we generate a StepFailureReport from it and clean up the running nicely, instead of showing the user a stack trace that they might not be able to parse very well. So, the exception is only re-raised if we are in debugging mode